### ISSUE TITLE
MDBOOK002 false positives for valid relative links

### ISSUE THREAD

**Issue by joshrotenberg** [joshrotenberg/mdbook-lint#162]
## Description
MDBOOK002 (internal-link-validation) reports errors for links that resolve to existing files when those files use relative paths or are in different directory levels.

## Examples of False Positives
```
introduction.md:58:3: error: MDBOOK002/internal-link-validation: Internal link './getting-started/installation.md' does not resolve to an existing file
cli/examples-detailed.md:552:3: error: MDBOOK002/internal-link-validation: Internal link 'cli-reference/index.md' does not resolve to an existing file
getting-started/installation.md:133:3: error: MDBOOK002/internal-link-validation: Internal link 'quickstart.md' does not resolve to an existing file
```

## File Structure
```
docs/src/
├── introduction.md
├── getting-started/
│   ├── installation.md
│   └── quickstart.md
├── cli/
│   └── examples-detailed.md
└── cli-reference/
    └── index.md
```

All reported files exist at the specified paths.

## Root Cause Analysis
The link validation appears to have issues with:
1. Relative path resolution from different directory depths
2. Links without leading `./`
3. Links to parent directories with `../`
4. Links from nested directories to sibling directories

## Expected Behavior
Valid relative links to existing files should not be reported as errors.

## Environment
- mdbook-lint: v0.11.2
- mdbook: 0.4.52
- OS: macOS Darwin 24.6.0

## Test Repository
https://github.com/joshrotenberg/redisctl (branch: fix/reenable-mdbook-lint-issue-37)

## Workaround
Currently have to disable MDBOOK002 entirely:
```toml
disabled-rules = ["MDBOOK002"]
```
